Remove deadcode from SVGResourcesCache::clientStyleChanged

rwlbuis opened the pull request:
#### cd8b6c24edddb345943f0eca8c04a13a059f9672
<pre>
Remove deadcode from SVGResourcesCache::clientStyleChanged
<a href="https://bugs.webkit.org/show_bug.cgi?id=253403">https://bugs.webkit.org/show_bug.cgi?id=253403</a>

Reviewed by Nikolas Zimmermann.

Remove code that is never ran due to clientStyleChanged
only being called by SVG renderers which are only
associated with SVGElements, making the if statement
always false.

* Source/WebCore/rendering/svg/SVGResourcesCache.cpp:
(WebCore::SVGResourcesCache::clientStyleChanged):

Canonical link: <a href="https://commits.webkit.org/269518@main">https://commits.webkit.org/269518@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/39e63df0333977a06c1d390c5e25cafccd0cf74a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22356 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/22554 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23433 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24259 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20691 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/22609 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/26817 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22888 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21701 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22594 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/22288 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19408 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25114 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/19338 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20263 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26510 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20355 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20497 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24370 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/21027 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17819 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/20481 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5433 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/24716 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/21019 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann commented on Source/WebCore/rendering/svg/SVGResourcesCache.cpp:
```diff
@@ -108,6 +108,8 @@ static inline bool rendererCanHaveResources(RenderObject& renderer)
 
 void SVGResourcesCache::clientStyleChanged(RenderElement& renderer, StyleDifference diff, const RenderStyle* oldStyle, const RenderStyle& newStyle)
 {
+    ASSERT(!renderer.element() || renderer.element()->isSVGElement());
```
The first condition is also still suspicious :-) Do we call clientStyleChanged() for anonymous renderers?
I think it doesn't actually add much value to add this assertion here, no? We're no longer using renderer.element() in this function.

rwlbuis replied:
Don't we have a hack in RenderTreeUpdater where an anon SVG renderer is created?

nikolaszimmermann replied:
Yeah, the RenderSVGRoot first child is a RenderSVGViewportContainer associated with no element (anonymous renderer). However, it is not possible to apply resources to that specific renderer -- and since there are no other anonymous renderrs in the SVG subtree, this shouldn't be possible either (a call to clientStyleChanged() by an anonymous renderer).

rwlbuis replied:
ok, I need to debug a bit then since the condition is needed (at least for debug builds, otherwise there are crashes).

rwlbuis replied:
So, RenderTreeBuilder::SVG::createViewportContainer will call initializeStyle, which will call styleDidChange, which will call clientStyleChanged.

rwlbuis committed:
Remove deadcode from SVGResourcesCache::clientStyleChanged
https://bugs.webkit.org/show_bug.cgi?id=253403

Reviewed by Nikolas Zimmermann.

Remove code that is never ran due to clientStyleChanged
only being called by SVG renderers which are only
associated with SVGElements, making the if statement
always false.

* Source/WebCore/rendering/svg/SVGResourcesCache.cpp:
(WebCore::SVGResourcesCache::clientStyleChanged):

Canonical link: https://commits.webkit.org/269518@main
